engine/lite/embeddedDeploymentManager/src/main/scala/pl/touk/nussknacker/engine/embedded/RequestResponseDeploymentStrategy.scala

```diff
@@ -0,0 +1,106 @@
+import pl.touk.nussknacker.engine.api.deployment.simple.SimpleStateStatus
+import pl.touk.nussknacker.engine.api.process.ComponentUseCase
+import pl.touk.nussknacker.engine.api.{JobData, MetaData, RequestResponseMetaData}
+import pl.touk.nussknacker.engine.embedded.RequestResponseDeploymentStrategy.RequestResponseConfig
+import pl.touk.nussknacker.engine.graph.EspProcess
+import pl.touk.nussknacker.engine.lite.TestRunner
+import pl.touk.nussknacker.engine.lite.api.runtimecontext.LiteEngineRuntimeContextPreparer
+import pl.touk.nussknacker.engine.requestresponse.FutureBasedRequestResponseScenarioInterpreter.InterpreterType
+import pl.touk.nussknacker.engine.requestresponse.{FutureBasedRequestResponseScenarioInterpreter, RequestResponseEngine}
+import pl.touk.nussknacker.engine.resultcollector.ProductionServiceInvocationCollector
+import pl.touk.nussknacker.engine.util.config.ConfigEnrichments._
+
+import scala.collection.concurrent.TrieMap
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.{Await, ExecutionContext, Future}
+import scala.util.{Failure, Success, Try}
+
+object RequestResponseDeploymentStrategy {
+
+  case class RequestResponseConfig(port: Int, interface: String = "0.0.0.0")
+
+  def apply(config: Config)(implicit as: ActorSystem): RequestResponseDeploymentStrategy = {
+    new RequestResponseDeploymentStrategy(config.rootAs[RequestResponseConfig])
+  }
+
+}
+
+class RequestResponseDeploymentStrategy(config: RequestResponseConfig)(implicit as: ActorSystem)
+  extends DeploymentStrategy with LazyLogging {
+
+  private val akkaHttpSetupTimeout = 10 seconds
+
+  private val pathToInterpreter = TrieMap[String, ScenarioInterpreter]()
+
+  private var server: ServerBinding = _
+
+  override type ScenarioInterpreter = FutureBasedRequestResponseScenarioInterpreter.InterpreterType
+
+  override def open(): Unit = {
+    logger.info(s"Serving request-response on ${config.port}")
+
+    val route = new ProcessRoute(pathToInterpreter)
+
+    import as.dispatcher
+    implicit val materializer: Materializer = Materializer(as)
+
+    server = Await.result(
+      Http().newServerAt(
+      interface = config.interface,
+      port = config.port
+    ).bind(route.route), akkaHttpSetupTimeout)
+  }
+
+  override def close(): Unit = {
+    Await.result(server.terminate(akkaHttpSetupTimeout), akkaHttpSetupTimeout)
+  }
+
+
+  override def onScenarioAdded(jobData: JobData,
+                               modelData: ModelData,
+                               parsedResolvedScenario: EspProcess,
+                               contextPreparer: LiteEngineRuntimeContextPreparer)(implicit ec: ExecutionContext): Try[InterpreterType] = {
+    import pl.touk.nussknacker.engine.requestresponse.FutureBasedRequestResponseScenarioInterpreter._
+
+    val engineWithPath = pathForScenario(jobData.metaData).product(RequestResponseEngine[Future](parsedResolvedScenario, jobData.processVersion, contextPreparer, modelData, Nil,
```

Comment:
1. can u extract logic of creation of engine from this line? this `path product ...` is already complicated a bit. Adding engine creation logic doesn't help :/ it can be just a local variable  
2. also it is not a good moment to change Engine -> Interpreter like it is in streaming? engine is the wider thing I guess

Reply:
:+1: